src/Link.js

```diff
@@ -1,7 +1,10 @@
 import { h } from "hyperapp"
 
 function getOrigin(loc) {
-  return loc.protocol + "//" + loc.hostname + (loc.port ? ":" + loc.port : "")
+  var protocol = (loc.protocol && ':' !== loc.protocol) ? loc.protocol : location.protocol
```

Comment:
@atomita Could you please explain what's going on here? We should create a function to document whatever bug of browser inconsistency this is trying to patch.

Reply:
@jorgebucaran 
I reconfirmed this issue.
May be a issue of access by localhost or IP.
I confirmed that `document.getElementsByTagName('a')[0].hostname` will be empty if I open host (http: 10.0.2.2: 9081 or http://localhost:9081) on IE 11 from windows 7 using "Modern.IE".

But, trying a small code with planker, the protocol and hostname seem to be fine. (Port has a issue)
http://plnkr.co/edit/PRtqM2AW0Daq3CbJvSO8